python/semantic_kernel/kernel.py

```diff
@@ -133,6 +134,110 @@ def register_semantic_function(
+                    variables = ContextVariables(input_str)
+                elif input_str is None and input_vars is not None:
+                    variables = input_vars
+                elif input_str is not None and input_vars is not None:
+                    variables = ContextVariables(input_str)
+                    variables = variables.merge_or_overwrite(
+                        new_vars=input_vars, overwrite=False
+                    )
+                else:
+                    variables = ContextVariables()
+                context = SKContext(
+                    variables,
+                    self._memory,
+                    self._skill_collection.read_only_skill_collection,
+                    self._log,
+                )
+        else:
+            raise ValueError("No functions passed to run")
+
+        try:
+            client: ChatCompletionClientBase | TextCompletionClientBase
+            client = stream_function._ai_service
+
+            # Get the closure variables from function for finding function_config
+            closure_vars = stream_function._function.__closure__
+            for var in closure_vars:
+                if isinstance(var.cell_contents, SemanticFunctionConfig):
+                    function_config = var.cell_contents
+                    break
+
+            if function_config.has_chat_prompt:
+                as_chat_prompt = cast(
+                    ChatPromptTemplate, function_config.prompt_template
+                )
+
+                # Similar to non-chat, render prompt (which renders to a
+                # list of <role, content> messages)
+                completion = ""
+                messages = await as_chat_prompt.render_messages_async(context)
+                async for steam_message in client.complete_chat_stream_async(
+                    messages, stream_function._chat_request_settings
+                ):
+                    completion += steam_message
+                    yield steam_message
+
+                # Add the last message from the rendered chat prompt
+                # (which will be the user message) and the response
+                # from the model (the assistant message)
+                _, content = messages[-1]
+                as_chat_prompt.add_user_message(content)
+                as_chat_prompt.add_assistant_message(completion)
+
+                # Update context
+                context.variables.update(completion)
+
+            else:
+                completion = ""
+                prompt = await function_config.prompt_template.render_async(context)
+                async for stream_message in client.complete_stream_async(
+                    prompt, stream_function._ai_request_settings
+                ):
+                    completion += stream_message
+                    yield stream_message
+                context.variables.update(completion)
+
+        except Exception as e:
+            # TODO: "critical exceptions"
```

Comment:
It will be very helpful to rise exception here.
If you import skills from file and your config does not contain "number_of_responses": 1 you will not get anything.

The error occurs in OpenAIChatCompletion at line 141 "request_settings.number_of_responses > 1"

Error ""'>' not supported between instances of 'NoneType' and 'int'""